### ISSUE TITLE
OpenSSL::X509::StoreError for repositories { rubygems() } under JDK 11

### ISSUE THREAD

**Issue by dmolesUC3** [jruby-gradle/jruby-gradle-plugin#323]
I had this project working with a manually installed gem, but when I try to use the rubygems Maven bridge, I get `OpenSSL::X509::StoreError: setting default path failed: No password supplied for PKCS#12 KeyStore.`

The full stack trace is pretty long (it's in a [gist here](https://gist.github.com/dmolesUC3/d344c7cb7c8dc0a6aa58371adb960b54)) but here's the first few lines:

```
OpenSSL::X509::StoreError: setting default path failed: No password supplied for PKCS#12 KeyStore.
    set_default_paths at org/jruby/ext/openssl/X509Store.java:165
           SSLContext at uri:classloader:/META-INF/jruby.home/lib/ruby/shared/jopenssl19/openssl/ssl-internal.rb:31
                  SSL at uri:classloader:/META-INF/jruby.home/lib/ruby/shared/jopenssl19/openssl/ssl-internal.rb:22
              OpenSSL at uri:classloader:/META-INF/jruby.home/lib/ruby/shared/jopenssl19/openssl/ssl-internal.rb:21
               (root) at uri:classloader:/META-INF/jruby.home/lib/ruby/shared/jopenssl19/openssl/ssl-internal.rb:20
                 load at org/jruby/RubyKernel.java:1087
```

## Build script

```groovy
buildscript {
  repositories {
    jcenter()
  }
  dependencies {
    classpath 'com.github.jruby-gradle:jruby-gradle-plugin:1.5.0'
  }
}

apply plugin: "com.github.jruby-gradle.base"

sourceCompatibility = 9
targetCompatibility = 9

jruby {
  defaultVersion '9.1.14.0'
}

ext {
  jrubyVersion = '9.1.14.0'
}

repositories {
  rubygems('https://rubygems.org')
}

dependencies {
  gems "rubygems:rdf-vocab:2.2.8"
  compile "org.jruby:jruby-complete:$jrubyVersion"

  // ... other deps ...
}
```

## Environment

OpenSSL:

```
LibreSSL 2.2.7
```

JRuby:

```
jruby 9.1.14.0 (2.3.3) 2017-11-08 2176f24 Java HotSpot(TM) 64-Bit Server VM 9.0.1+11 on 9.0.1+11 +jit [darwin-x86_64]
```

Gradle:

```
------------------------------------------------------------
Gradle 4.2.1
------------------------------------------------------------

Build time:   2017-10-02 15:36:21 UTC
Revision:     a88ebd6be7840c2e59ae4782eb0f27fbe3405ddf

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015
JVM:          9.0.1 (Oracle Corporation 9.0.1+11)
OS:           Mac OS X 10.13.2 x86_64
```

**Comment by rtyler** [comment]
I don't have a Mac or JDK9 to double check this, but it's definitely not broken on Linux with JDK8 :smile_cat:

**Comment by rtyler** [comment]
Welp, looks like I was able to reproduce this in a Docker image running OpenJDK 11.0.3 (`openjdk:11-stretch`).

:hankey:

**Comment by rtyler** [comment]
This is sort of reproducible in tests, although via a roundabout way.

 JRubyExecIntegrationSpec. Running rspec from a script should not cause a gemWorkDir failure 

```
io/console on JRuby shells out to stty for most operations
ERROR:  Loading command: install (OpenSSL::X509::StoreError)
	setting default path failed: No password supplied for PKCS#12 KeyStore.
ERROR:  While executing gem ... (NoMethodError)
    undefined method `invoke_with_build_args' for nil:NilClass
```

**Comment by rtyler** [comment]
This might be related to some upstream issues which is unfortunate:

See:

*  https://github.com/jruby/jruby/issues/5699

**Comment by rtyler** [comment]
So it looks like JRuby 9.2.1.0 should be fine, and our tests reflect this. 

I'm going to update our tests to make our "old" JRuby version that we test with, 9.2.1.0 and add some notes about JDK9+ support.

**Comment by rtyler** [comment]
@dmolesUC3 so this should be fixed with JRuby/Gradle 1.7.0 and later, which default to JRuby 9.2..7.0.

As of 2.0.0 we will stop supporting older versions of things, and start testing against JDK11 making sure issues like this do not come up.
